### PR TITLE
feat: actor state accessors: Add accessors to DealState.SectorNumber

### PR DESCRIPTION
**Pull request by magik6k**
## Proposed Changes
Prompted by the fact that this will be really useful for me in RIBS, and very likely for others - it is now much, much easier to match on-chain deals with sectors.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Review by LexLuthr — APPROVED** [review]

**Review by rvagg — APPROVED** [review]
Nice. I think I recall a conversation during the migration work that `0` is a valid sector number, that's not an active concern here is it? Would -1 be a better signal?

File: api/api_full.go [hunk 1]
```diff
@@ -1166,9 +1166,10 @@ type MarketBalance struct {
...
-	SectorStartEpoch abi.ChainEpoch // -1 if not yet included in proven sector
-	LastUpdatedEpoch abi.ChainEpoch // -1 if deal state never updated
-	SlashEpoch       abi.ChainEpoch // -1 if deal never slashed
+	SectorStartEpoch abi.ChainEpoch   // -1 if not yet included in proven sector
+	LastUpdatedEpoch abi.ChainEpoch   // -1 if deal state never updated
+	SlashEpoch       abi.ChainEpoch   // -1 if deal never slashed
+	SectorNumber     abi.SectorNumber // 0 if not activated or not supported (pre-nv22)
 }
 
 func MakeDealState(mds market.DealState) MarketDealState {
```
Comment (rvagg) [inline api/api_full.go:1175]: this is missing `SectorNumber` in the constructor

**Comment by rvagg** [comment]
I accidentally did the same PR at https://github.com/filecoin-project/lotus/pull/11998, but I have the minor fix I mentioned above; perhaps someone could review that PR and we can just land it. Or fix this one and land it, either way is fine.

**Comment by rvagg** [comment]
merged #11998

**Pull request closed without merging**